src/Forms/abstractbilinearform.jl

```diff
@@ -0,0 +1,14 @@
+abstract type AbstractForm{V,BraMap,KetMap,OperatorMap,SpaceMap} <:
```

Comment:
I don't think there is a need for `AbstractForm` to have so many type parameters. Abstract types generally have fewer parameters than the subtypes, and just try to have ones which are expected to be common across all or most subtypes. We could even just start with it just having `V` to match `AbstractITensorNetwork{V}`.